server/src/main/java/org/elasticsearch/common/collect/StreamUtils.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.collect;
+
+import java.util.stream.Stream;
+
+public class StreamUtils {
+
+    private StreamUtils() {
+        // no instances
+    }
+
+    @SafeVarargs
+    @SuppressWarnings("varargs")
+    public static <T> Stream<T> concat(Stream<T>... streams) {
+        if (streams.length == 0) {
```

Comment:
Why not just `return Arrays.stream(streams).flatMap(Function.identity());` ? We might even get into stack depth troubles if we use the concat like we do here?
The docs for concat state:

```
Use caution when constructing streams from repeated concatenation. Accessing an element of a deeply concatenated stream can result in deep call chains, or even StackOverflowError.
Subsequent changes to the sequential/parallel execution mode of the returned stream are not guaranteed to be propagated to the input streams.
```

Reply:
> We might even get into stack depth troubles

Touché :grin:

Yeah you're right `.flatMap(Function.identity())` is better here, no need for a utility to do this.